sceptre/template.py

```diff
@@ -127,97 +130,70 @@ def _call_sceptre_handler(self):
             sys.path.remove(os.path.join(os.getcwd(), directory))
         return body
 
-    def upload_to_s3(
-            self, region, bucket_name, key_prefix, environment_path,
-            stack_name, connection_manager
-    ):
+    def upload_to_s3(self):
         """
         Uploads the template to ``bucket_name`` and returns its URL.
 
         The template is uploaded with the key
         ``<key_prefix>/<region>/<environment_path>/<stack_name>-<timestamp>.json``.
 
-        :param region: The AWS region to create the bucket in.
-        :type region: str
-        :param bucket_name: The name of the bucket to create.
-        :type bucket_name: str
-        :param key_prefix: A string to prefix to the key used to store the
-            template in S3.
-        :type key_prefix: str
-        :param environment_path: The environment that the stack belongs to.
-        :type env_path: str
-        :param stack_name: The name of the stack that this template creates.
-        :type stack_name: str
-        :param connection_manager: The connection manager used to make
-            AWS calls.
-        :type connection_manager: sceptre.connection_manager.ConnectionManager
         :returns: The URL of the template object in S3.
         :rtype: str
         :raises: botocore.exceptions.ClientError
 
         """
+        if not self.s3_details:
+            raise Exception()
+
         self.logger.debug("%s - Uploading template to S3...", self.name)
 
         with self._boto_s3_lock:
-            if not self._bucket_exists(bucket_name, connection_manager):
-                self._create_bucket(region, bucket_name, connection_manager)
+            if not self._bucket_exists():
+                self._create_bucket()
```

Comment:
Does there need to be error handling for when a bucket doesn't exist but a chosen name is already taken?

Reply:
Agree! Think however that would be new functionality - will do a separate PR.

Reply:
Ah yes, that's definitely outside the scope of a "refactor" ticket.